### ISSUE TITLE
docker-personality logs are filled (MBs) with - INFO Got an error trying to convert public port number to an int

### ISSUE THREAD

**Issue by malikkal** [vmware/vic#7117]
VIC Engine v1.3.0-15556-473375a

After a VCH reset, you could encounter the above and following docker operations fails.

docker stop would fail with below:
`Error response from daemon: all ports are allocated`

while docker kill would successfully power off the cVM, we receive an error as below.
```
Error response from daemon: Cannot kill container 7f1c3f8f9295: all ports are allocated
Error response from daemon: Cannot kill container 78de13767fea: all ports are allocated
```
VCH operations could be brought back to normal only after a docker kill and reboot of VCH. Subsequently we **must remove** the powered of cVMs using docker rm after docker-personality is up. Should you try to start the stopped cVMs, we would risk going into the same episode starting with

`Error response from daemon: all ports are allocated.`

I am not sure what triggers the whole episode. But, quite sure, we never encountered this on older versions.

This is probably related to #7084. However, I originally thought they are only applicable during an upgrade and cVMs left running from the previous VCH versions. But, that doesn't seems like it.

**Comment by AngieCris** [comment]
Cc: @hickeng This seems to be related to all ports being forwarded for container on open network

**Comment by hickeng** [comment]
@malikkal This is an issue introduced in `1.3.0` that impacts only containers on Open networks and is triggered by restart. One of the docker libraries is taking the port range 0-65535 which are the accessible ports on an open network and decomposing that into a port mapping for each port - this is what you identified in #7091.
When the endpointVM restarts it reapplies NAT forwarding for running containerVMs and _mistakenly_ attempts to apply a NAT rule for each of those 65535 ports, thereby exhausting the port space.

The application of those NAT rules involves a lot of iteration, many invocations of iptables, and results in a large logging overhead - I _believe_ but have not confirmed that this is what caused your initial upgrade failures; it was actually just taking that long to come up because of the number of port mappings being reconstructed. This portion needs confirming.

Upgrade hits this specifically because it's synonymous with a restart of a `1.3.0` endpointVM.

**Comment by hickeng** [comment]
Working on #7084 which should address this as well. Kept issues separate for now to ensure we confirm the distinct symptoms noted.

**Comment by hickeng** [comment]
Closed by #6641 - this was left open until now as #6641 exposed an issue that #7140 addresses. Both are now merged.
